jax/_src/interpreters/ad.py

```diff
@@ -387,16 +387,23 @@ def process_custom_jvp_call(self, _, __, f_jvp, tracers, *, symbolic_zeros):
   def post_process_custom_jvp_call(self, out_tracers, _):
     raise CustomJVPException()
 
-  def process_custom_vjp_call(self, _, __, fwd, bwd, tracers, out_trees):
+  def process_custom_vjp_call(self, _, __, fwd, bwd, tracers, out_trees,
+                              symbolic_zeros):
     primals_in, tangents_in = unzip2((t.primal, t.tangent) for t in tracers)
-    tangents_in = map(instantiate_zeros, tangents_in)
-    res_and_primals_out = fwd.call_wrapped(*map(core.full_lower, primals_in))
+    fwd_in = [(core.full_lower(p), type(t) is Zero)
+              for p, t in zip(primals_in, tangents_in)]
+    fwd_in = [x for pair in fwd_in for x in pair]   # flatten
+    res_and_primals_out = fwd.call_wrapped(*fwd_in)
     out_tree, res_tree = out_trees()
     res, primals_out = split_list(res_and_primals_out, [res_tree.num_leaves])
     avals_out = [raise_to_shaped(core.get_aval(x)) for x in primals_out]
+    # We don't need to handle any symbolic zeros on tangents_in or
+    # tangents_out below, because custom_lin_p is never executed and
+    # doesn't correspond to any custom user rule.
+    tangents_in = map(instantiate_zeros, tangents_in)
```

Comment:
It'd be nice to avoid instantiating zeros here... can we just prune tangents_in and bind custom_lin_p to the nonzeros?

Reply:
Seems like we could do that and then just adapt `_custom_lin_transpose` to handle the potentially shorter list of return values.

Reply:
Actually, neither do I in hindsight! Let's take care of this in a new PR.